src/terrain/terrain.js

```diff
@@ -401,10 +398,6 @@ export class Terrain extends Elevation {
             this._emptyDEMTextureDirty = !this._initializing;
         }
 
-        const options = this.painter.options;
-        this.renderCached = (options.zooming || options.moving || options.rotating || !!this.forceRenderCached) && !this._invalidateRenderCache;
```

Comment:
Does this logic still exist, just elsewhere? If so, I didn't succeed in tracking it down.

Reply:
Only the logic for this._invalidateRenderCache is kept. The rest is not necessary anymore and adds complexity to the number of possible code paths, it's a reason why the flicker is only visible during moving frames in #10763.

To provide some context, we launched v2 having two terrain render modes being fairly implicit:
- One render mode using the render cache during interactive frames, not preserving draw order
- One render mode using interleaved draw order otherwise (no caching even for large draped sequences)
The reason being that interleaved rendering was too expensive and honoring draw order on still fames was a fine workaround. The issue with this approach is that _still_ and _moving_ fames  were not rendering layers in the same order, leading to noticeable differences when the user stopped moving the camera for example.

In #10258, we moved away from this behavior to have an explicit opt-in render mode with optimizerForTerrain, allowing that decision to be based on the needs of the user depending on whether draw order is important, instead of being implicit to the renderer.

In draw order priority, we still do our best to identify sequences of draped batches to render them cached. Now, this particular check 'options.zooming || options.moving || options.rotating' is what determines _still_ vs _moving_ frames, but we can simply render cache always as it doesn't have any effect anymore: with or without draw order we are always rendering the _same frames_.